lib/packetgen/types/http_headers.rb

```diff
@@ -0,0 +1,59 @@
+# This file is part of PacketGen
+# See https://github.com/sdaubert/packetgen for more informations
+# Copyright (C) 2016 Sylvain Daubert <sylvain.daubert@laposte.net>
+# This program is published under MIT license.
+
+module PacketGen
+  module Types
+
+    # @abstract Base class for HTTP headers.
+    # @author Kent 'picat' Gruber
+    class HTTPHeaders
```

Comment:
This class should only be used by HTTP, so you should define it in HTTP namespace, not in Types one